### ISSUE TITLE
kcov not covering case statements with balanced parenthesis

### ISSUE THREAD

**Issue by Mark-E-Hamilton** [SimonKagstrom/kcov#62]
We use balanced parenthesis in all of our shell case statement so that syntax hightlighting and brace matching works correctly in editors. However, if I test this example script four times, passing it  'a', 'b', 'c', and 'd' successively :
````
#!/bin/bash
case ${1} in
(a)
   echo Got $1
   ;;
(b)
   echo Got $1
   ;;
c)
   echo Got $1
   ;;
*)
   echo Got $1
   ;;
esac
````
kcov only reports 71.4%  coverage. The case and echo statements are all green, the 'c0' and '*0' lines are white (not executed, I think) but the 'a' and '(b)' lines are red for not covered

**Comment by SimonKagstrom** [comment]
OK, the bug is caused by this code in bash-engine.cc:

```c
			// Case switches are nocode
			if (caseActive && s[s.size() - 1] == ')') {
				// But let functions be
				if (s.find("(") == std::string::npos)
					continue;
			}
```

I'll test more tonight, but probably it should be enough to have the function-find-test be false if the line starts with a parenthesis.

I've been hoping to avoid having to write a proper yacc parser for bash, but perhaps it will be needed after all...

**Comment by SimonKagstrom** [comment]
Fixed with b8c3ad9

**Comment by Mark-E-Hamilton** [comment]
Impressive response time. I've pulled it, and it works. Thanks.
